src/prefect/infrastructure/kubernetes.py

```diff
@@ -625,24 +641,23 @@ def _watch_job(self, job_name: str) -> int:
                         exc_info=True,
                     )
 
-        self.logger.debug(f"Job {job_name!r}: Starting watch for job completion")
-        deadline = (
-            (time.time() + self.job_watch_timeout_seconds)
-            if self.job_watch_timeout_seconds is not None
-            else None
-        )
-        completed = False
-        while not completed:
-            remaining_time = deadline - time.time() if deadline else None
-            if deadline and remaining_time <= 0:
-                self.logger.error(
-                    f"Job {job_name!r}: Job did not complete within "
-                    f"timeout of {self.job_watch_timeout_seconds}s."
-                )
-                return -1
+        with self.get_batch_client() as batch_client:
```

Comment:
This section perhaps best read without whitespace changes https://github.com/PrefectHQ/prefect/pull/8618/files?diff=unified&w=1